File: leet_rust/src/contains_duplicate.rs
use std::collections::HashSet;

pub fn contains_duplicate(nums: Vec<i32>) -> bool {
    let mut unique_vals = HashSet::new();
    for num in &nums {
        if unique_vals.contains(&num) {
            return true;
        }
        unique_vals.insert(num);
    }
    false
}

File: contains_duplicates/Python/containsDuplicate.py
# 217. Contains Duplicate 

# Input: nums = [1,2,3,1]
# Output: true

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        hashset = set()

        for n in nums:
            if n in hashset:
                return True
            hashset.add(n);

        return False
